packages/p2p-v2/src/routes/AppContent/index.tsx

```diff
@@ -13,31 +14,58 @@ const tabRoutesConfiguration = routes.filter(route => route.name !== 'Advertiser
 const AppContent = () => {
     const history = useHistory();
     const location = useLocation();
-    const { data: activeAccountData, isLoading } = useActiveAccount();
+    const { data: activeAccountData, isLoading: isLoadingActiveAccount } = useActiveAccount();
+    const { isSuccess } = useAuthorize();
 
     const getActiveTab = (pathname: string) => {
         const match = routes.find(route => pathname.startsWith(route.path));
         return match ? match.name : BUY_SELL_URL;
     };
 
     const [activeTab, setActiveTab] = useState(() => getActiveTab(location.pathname));
-    const { subscribe } = p2p.settings.useGetSettings();
+    const [hasCreatedAdvertiser, setHasCreatedAdvertiser] = useState(false);
+    const { subscribe: subscribeP2PSettings } = p2p.settings.useGetSettings();
+    const { error, isIdle, isLoading, isSubscribed, subscribe: subscribeAdvertiserInfo } = p2p.advertiser.useGetInfo();
 
     useEffect(() => {
-        if (activeAccountData) subscribe();
-    }, [activeAccountData, subscribe]);
+        if (activeAccountData) {
+            subscribeP2PSettings();
+        }
+    }, [activeAccountData, subscribeP2PSettings]);
+
+    useEffect(() => {
+        if (isSuccess) {
+            subscribeAdvertiserInfo();
+        }
+    }, [isSuccess, subscribeAdvertiserInfo]);
+
+    // Need this to subscribe to advertiser info after user has created an advertiser.
+    // setHasCreatedAdvertiser is triggered inside of NicknameModal.
+    useEffect(() => {
+        if (isSuccess && hasCreatedAdvertiser) {
+            subscribeAdvertiserInfo();
+        }
+    }, [hasCreatedAdvertiser, isSuccess, subscribeAdvertiserInfo]);
```

Comment:
When the user has not added their nickname yet, and are already POA + POI verified, the error from `p2p_advertiser_info` api endpoint will return `AdvertiserNotFound`. Because of this, the first useEffect's subscribe function will try to subscribe, but then it will fail. So the second useEffect is a must as when the user creates their account for p2p, we will trigger the above state `hasCreatedAdvertiser` to subscribe to the newly created user.

Prior we would use invalidation, but we can't do that for useSubscription